### DISCUSSION TITLE
reset a service after test

### DISCUSSION THREAD

**Discussion by Tallyb** [help-me-mom/ng-mocks#7664] [General]
beforeEach(() => MockBuilder(MyStore));

  beforeEach(() => {
    store = ngMocks.get(MyStore);
});

afterEach(() => ??? );

store is a state manager, and the state is kept between tests. 
Is there an elegant way to get a fresh instance of the service?

**Comment by satanTime** [comment]
Hi @Tallyb, could you give an example of the state you would like to reset?

I assume the test is using [`ngMocks.faster()`](https://ng-mocks.sudo.eu/api/ngMocks/faster), if yes, you should use something like: https://ng-mocks.sudo.eu/api/ngMocks/faster#example-of-optimizing-spies-in-beforeeach.

[`ngMocks.faster()`](https://ng-mocks.sudo.eu/api/ngMocks/faster) skips initialization of `TestBed` for the same setup, therefore all services and components keep state between tests without manual reset in `beforeEach` for example.
